src/webapi/formdata.rs

```diff
@@ -0,0 +1,207 @@
+use crate::Value;
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::blob::Blob;
+use webapi::element::Element;
+
+/// The `FormData` interface provides a way to easily construct a set of key/value pairs
+/// representing form fields and their values, which can then be easily sent using the
+/// `XMLHttpRequest.send()` method. It uses the same format a form would use if the encoding type
+/// were set to `"multipart/form-data"`.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/FormData)
+// https://xhr.spec.whatwg.org/#formdata
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "FormData")]
+pub struct FormData( Reference );
+
+impl FormData {
+    /// Creates a new `FormData`.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData)
+    pub fn new( form: &Element ) -> Self {
+        js! (
+            return new FormData(@{form});
+        ).try_into().unwrap()
+    }
+
+    /// Appends a new value onto an existing key, or adds the key if it does not already exist.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append)
+    // https://xhr.spec.whatwg.org/#dom-formdata-append
+    pub fn append( &self, name: &str, value: &str ) {
```

Comment:
I'd probably prefer to call it `append_string` since none of the overloads of `append`s really feel like the default one.